Assets/Scripts/Wall/WallSound.cs

```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+public class WallSound : MonoBehaviour
+{
+    [SerializeField] private AudioClip _dangerSound;
+
+    private readonly float _defaultVolume = 0.2f;
+    private AudioSource _source;
+
+    private void Start()
+    {
+        _source = gameObject.AddComponent<AudioSource>();
+    }
+
+    public void PlayDangerSound()
+    {
+        _source.PlayOneShot(_dangerSound, _defaultVolume);
```

Comment:
We can use `AudioSource.PlayClipAtPoint(_dangerSound, Camera.main.transform.position, _defaultVolume)`. Such we avoid creation of AudioSource.